rotkehlchen/tests/unit/test_unknown_asset.py

```diff
@@ -0,0 +1,158 @@
+from rotkehlchen.typing import ChecksumEthAddress
+
+
+SHUF_ID = 'SHUF'
+SHUF_ETHEREUM_ADDRESS = ChecksumEthAddress(
+    HexAddress(HexStr('0x3A9FfF453d50D4Ac52A6890647b823379ba36B9E')),
+)
+SHUF_NAME = 'Shuffle.Monster V3'
+SHUF_DECIMALS = 18
+
+
+class TestInitialization:
+
+    def test_default(self):
+        ue_token = UnknownEthereumToken(
+            identifier=SHUF_ID,
+            ethereum_address=SHUF_ETHEREUM_ADDRESS,
+        )
+        assert ue_token.identifier == SHUF_ID
+        assert ue_token.ethereum_address == SHUF_ETHEREUM_ADDRESS
+        assert ue_token.name is None
+        assert ue_token.decimals is None
+
+
+class TestPostInitialization:
+
+    def test_symbol(self):
+        ue_token = UnknownEthereumToken(
+            identifier=SHUF_ID,
+            ethereum_address=SHUF_ETHEREUM_ADDRESS,
+        )
+        assert ue_token.symbol == ue_token.identifier
+
+
+class TestOperators:
+
+    def test_eq(self):
+        ue_token_1 = UnknownEthereumToken(
+            identifier=SHUF_ID,
+            ethereum_address=SHUF_ETHEREUM_ADDRESS,
+            name=SHUF_NAME,
+            decimals=SHUF_DECIMALS,
+        )
+        ue_token_2 = UnknownEthereumToken(
+            identifier=SHUF_ID,
+            ethereum_address=SHUF_ETHEREUM_ADDRESS,
+            name=SHUF_NAME + 'name does no matter',
+            decimals=SHUF_DECIMALS - 10,
+        )
+        assert ue_token_1 == ue_token_2
+
+    def test_eq_invalid_type(self):
+        ue_token = UnknownEthereumToken(
+            identifier=SHUF_ID,
+            ethereum_address=SHUF_ETHEREUM_ADDRESS,
+        )
+        token = EthereumToken('WETH')
+
+        with pytest.raises(TypeError):
+            assert ue_token == token
+
+    def test_hash(self):
+        ue_token = UnknownEthereumToken(
+            identifier=SHUF_ID,
+            ethereum_address=SHUF_ETHEREUM_ADDRESS,
+        )
+        assert hash(ue_token) == hash((ue_token.identifier, ue_token.ethereum_address))
+
+    def test_ne(self):
+        ue_token_1 = UnknownEthereumToken(
+            identifier=SHUF_ID,
+            ethereum_address=SHUF_ETHEREUM_ADDRESS,
+        )
+        ue_token_2 = UnknownEthereumToken(
+            identifier=SHUF_ID.lower(),
+            ethereum_address=SHUF_ETHEREUM_ADDRESS,
+        )
+        assert ue_token_1 != ue_token_2
+
+
+class TestRepresentation:
+
+    def test_str_eq_name(self):
+        ue_token = UnknownEthereumToken(
+            identifier=SHUF_ID,
+            ethereum_address=SHUF_ETHEREUM_ADDRESS,
+            name=SHUF_NAME,
+            decimals=SHUF_DECIMALS,
+        )
+        assert str(ue_token) == ue_token.name
+
+    def test_str_eq_identifier(self):
+        ue_token = UnknownEthereumToken(
+            identifier=SHUF_ID,
+            ethereum_address=SHUF_ETHEREUM_ADDRESS,
+            decimals=SHUF_DECIMALS,
+        )
+        assert str(ue_token) == ue_token.identifier
+
+    def test_repr(self):
```

Comment:
A comment on the general direction. Thanks for even writing a test for repr. But generally in the future no need to overdo it with too many tests. I don't care about silly things like 100% coverage. Yes good test coverage is good. But we also need to work fast and we can only do that when we write just the amount of tests necessary to know our stuff work.

I am pretty sure that this test did not take almost any time for you to write. But after seeing the amount of stuff that is unit tested I just felt like mentioning it. So as a general guideline If you at some point see that writing too many unit tests slows down the release of a feature I expect you to switch immediately out of testing and keep on implementing the required work for the feaure.

Reply:
Got it